shell/platform/windows/text_input_plugin.cc

```diff
@@ -43,8 +43,7 @@ void TextInputPlugin::CharHook(Win32FlutterWindow* window,
   if (active_model_ == nullptr) {
     return;
   }
-  // TODO bug 30661
-  active_model_->AddCharacter(static_cast<char>(code_point));
+  active_model_->AddCharacter(code_point);
```

Comment:
The `code_point` argument should change to `char32_t` as well. We should push the specific type all the way through to the point where we're interfacing with the Win32 code rather than passing around a generic unsigned int.

Reply:
done.